### PR TITLE
Showing a specific doc section based for books and journals

### PR DESCRIPTION
**Pull request by musangowope**
I started adding the functionality to replicate the functionality we have on `edit.indigo`. 
I.e on click of table of content item, show only the corresponding part of the document.

This document behavior will be applicable to a specific document type books and journals

DEMO:
https://www.loom.com/share/a9475156e8b442818e723c0b1cecccc2

### CODE REVIEW

File: peachjam/js/components/DocumentContent/index.ts [hunk 1]
```diff
@@ -30,18 +30,25 @@ class DocumentContent {
   private searchApp: any;
   private navOffCanvas: OffCanvas | undefined;
   private enchrichmentsManager: EnrichmentsManager | null;
+  private storedDocument: Node | undefined;
+  private tocController: HTMLElement| null;
   constructor (root: HTMLElement) {
     this.root = root;
     this.navOffCanvas = undefined;
     this.enchrichmentsManager = null;
+    this.storedDocument = this.root.querySelector('[data-document-element]')?.cloneNode(true);
```
Comment (longhotsummer) [inline peachjam/js/components/DocumentContent/index.ts:39]: You run this `querySelector` three times. Better to do it here in the constructor once and store the element for re-use.

File: peachjam/js/components/DocumentContent/index.ts [hunk 2]
```diff
@@ -147,16 +153,38 @@ class DocumentContent {
...
+      if (id) {
+        const sectionOfFocus = this.storedDocument.querySelector(`#${id}`)?.cloneNode(true) as HTMLElement | undefined;
+        if (!sectionOfFocus) return;
+        // Delete content within document element and then append section of focus
+        documentElement.replaceChildren(sectionOfFocus);
+      } else {
+        const storedDocument = this.storedDocument.cloneNode(true);
+        // @ts-ignore
+        documentElement.replaceChildren(...this.storedDocument.children);
+        this.storedDocument = storedDocument;
```
Comment (longhotsummer) [inline peachjam/js/components/DocumentContent/index.ts:172]: I don't think this is necessary?

File: peachjam/js/components/DocumentContent/index.ts [hunk 3]
```diff
@@ -147,16 +153,38 @@ class DocumentContent {
...
+      }
+      this.enchrichmentsManager?.layoutItems();
```
Comment (longhotsummer) [inline peachjam/js/components/DocumentContent/index.ts:174]: Is this required? The gutter watches the main container and reflows when its children change, which should be sufficient.

File: peachjam/js/components/DocumentContent/index.ts (hunk 1 above)
Comment (musangowope) [reply peachjam/js/components/DocumentContent/index.ts]: done

File: peachjam/js/components/DocumentContent/index.ts (hunk 2 above)
Comment (musangowope) [reply peachjam/js/components/DocumentContent/index.ts]: refactored

File: peachjam/js/components/DocumentContent/index.ts (hunk 3 above)
Comment (musangowope) [reply peachjam/js/components/DocumentContent/index.ts]: good catch, nope this is not required. refactored

File: peachjam/js/components/DocumentContent/index.ts [hunk 4]
```diff
@@ -30,18 +30,27 @@ class DocumentContent {
...
+  private documentElement: Element | null;
   constructor (root: HTMLElement) {
     this.root = root;
     this.navOffCanvas = undefined;
     this.enchrichmentsManager = null;
+    this.documentElement = this.root.querySelector('[data-document-element]');
+    this.originalDocCloned = this.documentElement?.cloneNode(true);
+    this.tocController = this.setupTocForTab();
+
+    if (root.dataset.frbrUriSubtype === 'book' && root.getAttribute('data-display-type') === 'akn') {
```
Comment (longhotsummer) [inline peachjam/js/components/DocumentContent/index.ts:44]: Please delegate this to an attribute on `root` -- that makes it easier for us to determine in python to en/disable this behaviour, and means this component doesn't need to know about subtypes etc.

File: peachjam/js/components/DocumentContent/index.ts [hunk 5]
```diff
@@ -147,16 +154,34 @@ class DocumentContent {
...
+    this.tocController?.addEventListener('itemTitleClicked', (e) => {
+      const customEvt = e as CustomEvent;
+      const id = customEvt.detail.target.getAttribute('href').replace('#', '');
+      if (!this.documentElement) return;
+      if (!(this.originalDocCloned && this.originalDocCloned instanceof HTMLElement)) return;
+      if (id) {
+        const sectionOfFocus = this.originalDocCloned.querySelector(`#${id}`)?.cloneNode(true) as HTMLElement | undefined;
+        if (!sectionOfFocus) return;
+        // Delete content within document element and then append section of focus
+        this.documentElement.replaceChildren(sectionOfFocus);
```
Comment (longhotsummer) [inline peachjam/js/components/DocumentContent/index.ts:167]: TIL about replaceChildren :)

File: peachjam/js/components/DocumentContent/index.ts (hunk 4 above)
Comment (longhotsummer) [reply peachjam/js/components/DocumentContent/index.ts:44]: eg. `data-toc-show-active-item-only` or something. If that's present, then this behaviour is turned on.

Comment (musangowope) [reply peachjam/js/components/DocumentContent/index.ts]: done

**Comment by longhotsummer** [comment]
Note to self: for HTML documents where we generate our own TOC based on headings, we need to "indent and group" the heading with its content (everything until the next heading).

**Review by longhotsummer — APPROVED** [review]

**Pull request merged**